src/achievements/AchievementRegistry.ts

```diff
@@ -0,0 +1,46 @@
+import { BotgartClient } from "../BotgartClient";
+import * as Util from "../Util";
+import { Achievement } from "./Achievement";
+
+//basically the type of the constructor function
+type AchievementType = new (client: BotgartClient) => Achievement<any>
+
+const achievementCreators: AchievementType[] = [];
+
+export function registerAchievement() {
```

Comment:
This is a cool idea! But from what I read in [the docs](https://www.typescriptlang.org/docs/handbook/decorators.html), (and tested) this could be simplified to:

```typescript
export function registerAchievement(target: AchievementType) {
    achievementCreators.push(target);
}
```
Sadly, arrow-functions can not be used in this case.

Decorators on the achievements would then be just:

```
@registerAchievement
export class ...
```
without the parantheses.

I would also suggest renaming the decorator to just `achievement` or `registerable`, or something like that, as decorators – in my experience – come as nouns or adjectives.

Reply:
The idea is, that using the arrow function yo are able to provide addtional "metadata" with the the decorator:

```
@registerAchievement(enabled = false)
export class ...
```

Reply:
Seeing that the current implementation does not take any parameters: do you have any particular parameters in mind? 🤔 If not, then that sounds like YAGNI for now.

Reply:
changed